### PR TITLE
CLOUDP-131649: Provide support for Chocolatey package manager

### PR DESCRIPTION
**Pull request by Razeer123**
## Proposed changes

Adds script for automatic generation of .nupkg Chocolatey package of Atlas CLI. 

_Jira ticket:_ [CLOUDP-131649](https://jira.mongodb.org/browse/CLOUDP-131649)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by fmenezes** [comment]
looks like mongocli nupkg is being generated as `atlascli.1.25.0.nupkg` are we thinking about fixing it for both or maybe only generating it for atlascli?

File: build/package/generate-msi.sh [hunk 1]
```diff
@@ -50,3 +50,9 @@ env GOOS=windows GOARCH=amd64 go build \
 go-msi make --path "${WIX_MANIFEST_FILE}"  --msi "dist/${PACKAGE_NAME}" --version "${VERSION}"
 
 go run ./tools/sign -file "dist/${PACKAGE_NAME}"
+
+if [[ "${TOOL_NAME:?}" == atlascli ]]; then
+  go run ./tools/update-chocolatey-package/update-chocolatey-package.go "${VERSION}" ./dist/"${PACKAGE_NAME}" https://fastdl.mongodb.org/mongocli/"${PACKAGE_NAME}"
+  cd ./build/package/chocolatey/temp
+  choco pack
```
Comment (fmenezes) [inline build/package/generate-msi.sh:57]: nit: I'd probably have called it from go

File: build/package/generate-msi.sh [hunk 2]
```diff
@@ -50,3 +50,9 @@ env GOOS=windows GOARCH=amd64 go build \
 go-msi make --path "${WIX_MANIFEST_FILE}"  --msi "dist/${PACKAGE_NAME}" --version "${VERSION}"
 
 go run ./tools/sign -file "dist/${PACKAGE_NAME}"
+
+if [[ "${TOOL_NAME:?}" == atlascli ]]; then
+  go run ./tools/update-chocolatey-package/update-chocolatey-package.go "${VERSION}" ./dist/"${PACKAGE_NAME}" https://fastdl.mongodb.org/mongocli/"${PACKAGE_NAME}"
```
Comment (fmenezes) [inline build/package/generate-msi.sh:55]:
```suggestion
  go run ./tools/chocolateypkg/main.go "${VERSION}" ./dist/"${PACKAGE_NAME}" https://fastdl.mongodb.org/mongocli/"${PACKAGE_NAME}"
```

File: build/package/generate-msi.sh (hunk 1 above)
Comment (Razeer123) [reply build/package/generate-msi.sh]: Done!

File: build/package/generate-msi.sh [hunk 3]
```diff
@@ -52,7 +52,5 @@ go-msi make --path "${WIX_MANIFEST_FILE}"  --msi "dist/${PACKAGE_NAME}" --versio
 go run ./tools/sign -file "dist/${PACKAGE_NAME}"
 
 if [[ "${TOOL_NAME:?}" == atlascli ]]; then
-  go run ./tools/update-chocolatey-package/update-chocolatey-package.go "${VERSION}" ./dist/"${PACKAGE_NAME}" https://fastdl.mongodb.org/mongocli/"${PACKAGE_NAME}"
-  cd ./build/package/chocolatey/temp
-  choco pack
+  go run ./tools/chocolateypkg/main.go "${VERSION}" ./dist/"${PACKAGE_NAME}" https://fastdl.mongodb.org/mongocli/"${PACKAGE_NAME}"
```
Comment (fmenezes) [inline build/package/generate-msi.sh:55]: nit: bonus points for flags
```suggestion
  go run ./tools/chocolateypkg/main.go -version "${VERSION}" -file ./dist/"${PACKAGE_NAME}" -url https://fastdl.mongodb.org/mongocli/"${PACKAGE_NAME}"
```

**Review by fmenezes — DISMISSED** [review]

File: build/package/generate-msi.sh (hunk 3 above)
Comment (Razeer123) [reply build/package/generate-msi.sh]: Cool idea. Implemented!

**Review by fmenezes — APPROVED** [review]

**Pull request merged**